src/bonds/sigaa-student-bond.ts

```diff
@@ -191,4 +199,65 @@ export class SigaaStudentBond implements StudentBond {
     }
     return listCourses;
   }
+
+  async getFrontPageActivities(): Promise<FrontPageActivities[]> {
+    const frontPage = await this.http.get(
+      '/sigaa/portais/discente/discente.jsf'
+    );
+    const table = frontPage.$('#avaliacao-portal > table');
+    const rows = table.find('tbody > tr');
+    const listActivities: FrontPageActivities[] = [];
+    for (const row of rows) {
+      const cellElements = frontPage.$(row).find('td');
+      const fullText = cellElements.text().replace(/(\r\n|\n|\r|\t)/gm, ' ');
+      const regex = /(\d{2}\/\d{2}\/\d{4})/g;
+      const matchesDate = regex.exec(fullText);
+      if (matchesDate) {
+        // regex para separar horario no padrão do sigaa dd/mm/yyyyHH:MM
+        const horario = fullText.trim().split(/(\d{2}\/\d{2}\/\d{4})/g)[2];
+        const regexHorario = /(\d{2}:\d{2})/g;
+        const matchesHorario = regexHorario.exec(horario);
+        // caso o matchesHorario não funcionar, ele coloca horario padrão
+        let hora = '23';
+        let minuto = '59';
+        if (matchesHorario) {
+          hora = matchesHorario[1].split(':')[0];
+          minuto = matchesHorario[1].split(':')[1];
+        }
+        const [dia, mes, ano] = matchesDate[1].split('/');
+        const dateObject = new Date(
+          `${parseInt(mes)}/${parseInt(dia)}/${parseInt(ano)} ${parseInt(
+            hora
+          )}:${parseInt(minuto)}`
+        );
+        const dateString = dateObject.toISOString();
+
+        const infoText = cellElements.find(' small').text();
```

Comment:
Acho que poderia usar `const infoText = this.parser.removeTagsHtml(cellElements.find('small').html());`
Dessa forma, você não precisaria usar os `.replace(/(\r\n|\n|\r|\t)/gm, '')` subsequentes
OBS, o removeTagsHtml só mantem as quebra de linha, então precisaria substitur os termos nos splits para o padrão `'\nTarefa:'`, `'\nAvaliação:'`, `'\nQuestionário:'`.
Outra coisa, será que não seria melhor pegar o infoText e dividir por linhas `.split('\n')` e na primeira linha pegar o nome da turma e na segunda linha dividir novamente só que com os dois pontos?
Assim, você terá o tipo de atividade e título dela, e se fosse de um tipo desconhecido, você poderia usar o `console.log` com um aviso dizendo que o tipo tal não foi identificado.